numpy/core/src/multiarray/iterators.c

```diff
@@ -1409,13 +1417,12 @@ arraymultiter_new(PyTypeObject *NPY_UNUSED(subtype), PyObject *args, PyObject *k
             ++n;
         }
     }
-    if (n < 1 || n > NPY_MAXARGS) {
-        if (PyErr_Occurred()) {
-            return NULL;
-        }
+    if (PyErr_Occurred()) {
```

Comment:
OK, do you see why this is there? I do not see how there can possibly an error set at this stage. But... It was there before, so I guess we can keep it (and moving it a bit doesn't matter much).

Reply:
Yes, isinstance can throw an exception in principle, when it returns -1

Reply:
That seems pretty far fetched, but OK. This whole thing is trying to be minimal invasive. And I think that is successful enough for me to be OK with merging it in a bit. (so if anyone has any stomach aches about it, please comment).

The C-API testing would be better of course, so maybe will also see to add that before merging.